pkg/ingester/errors.go

```diff
@@ -135,3 +136,95 @@ func formatMaxMetadataPerMetricError(limits *validation.Overrides, labels labels
 	))
 	return makeMetricLimitError(labels, err)
 }
+
+type ingesterErrSamplers struct {
+	sampleTimestampTooOld             *log.Sampler
+	sampleTimestampTooOldOOOEnabled   *log.Sampler
+	sampleTimestampTooFarInFuture     *log.Sampler
+	sampleOutOfOrder                  *log.Sampler
+	sampleDuplicateTimestamp          *log.Sampler
+	maxSeriesPerMetricLimitExceeded   *log.Sampler
+	maxMetadataPerMetricLimitExceeded *log.Sampler
+	maxSeriesPerUserLimitExceeded     *log.Sampler
+	maxMetadataPerUserLimitExceeded   *log.Sampler
+}
+
+func newIngesterErrSamplers(freq int64) ingesterErrSamplers {
+	if freq == 0 {
+		return ingesterErrSamplers{}
+	}
+	return ingesterErrSamplers{
+		sampleTimestampTooOld:             log.NewSampler(freq),
+		sampleTimestampTooOldOOOEnabled:   log.NewSampler(freq),
+		sampleTimestampTooFarInFuture:     log.NewSampler(freq),
+		sampleOutOfOrder:                  log.NewSampler(freq),
+		sampleDuplicateTimestamp:          log.NewSampler(freq),
+		maxSeriesPerMetricLimitExceeded:   log.NewSampler(freq),
+		maxMetadataPerMetricLimitExceeded: log.NewSampler(freq),
+		maxSeriesPerUserLimitExceeded:     log.NewSampler(freq),
+		maxMetadataPerUserLimitExceeded:   log.NewSampler(freq),
+	}
```

Comment:
As weird as this might look like, I would suggest doing:

```suggestion
	return ingesterErrSamplers{
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
		log.NewSampler(freq),
	}
```

That is valid style because the struct is not exported, and it's not hard to read because all fields are the same, but it has the benefit of not compiling if someone adds another sampler field but forgets to update the constructor (especially important because I suggested below to remove the methods and their tests)